ngrok/examples/mingrok.rs

```diff
@@ -24,5 +24,10 @@ async fn main() -> Result<(), Error> {
 
     info!(url = tun.url(), forwards_to, "started tunnel");
 
-    Ok(tun.forward_http(forwards_to).await?)
+    let fut = if forwards_to.contains('/') {
+        tun.forward_unix(forwards_to)
+    } else {
+        tun.forward_http(forwards_to)
+    };
+    Ok(fut.await?)
```

Comment:
Does this actually compile? :thinking: 

`forward_unix` and `forward_http` *should* return distinct unnameable `Future`-implementing types, and you aren't allowed to have the then/else branches of a conditional evaluate to different types. But then again, the sugar keeps getting sweeter and maybe I missed some ergonomics improvements :shrug:

Reply:
Compiles and runs! Had it proxying to my test process. VSCode says it's a `Pin<Box<dyn Future<Output = …> + Send>>`